server/room.mjs

```diff
@@ -122,6 +122,16 @@ export default class Room {
         player.send(func, args);
   }
 
+  deleteUnusedVariantFiles(stateID, oldState, newState) {
+    for(const oldVariantID in oldState.variants) {
+      if(!newState.variants[oldVariantID] && stateID.match(/^[a-z0-9]+$/) && oldVariantID.match(/^[a-z0-9]+$/)) {
```

Comment:
Consider moving the regexes into a single definition that's referenced everywhere instead of repeating them to make it easier to maintain & reduce the risk of updating in one place & not updating it other places.